### DISCUSSION TITLE
Analytical and FEMM

### DISCUSSION THREAD

**Discussion by nkbw33** [Eomys/pyleecan#703] [Q&A]
I have only just begun looking into Pyleecan for the lab I work in, but if I understand correctly it is capable of both analytical solving and FEMM. I was wondering if it is possible to isolate the analytical calculations and only use those? 

Am I understanding this correctly or are the analytical and FEMM implementations used for different parts of the calculation?

**Comment by BonneelP** [comment]
Hello,

Why are you looking to "isolate the analytical calculations" ? There are some analytical parts within pyleecan that could be reused but it depends of what you are looking for and what kind of simulation you plan to run. For FEMM, as long as your machine is defined in pyleecan format, you can draw and solve the flux density within FEMM.

Best regards,
Pierre

**Reply by nkbw33** [reply]
Sorry, I wasn't very clear, I will try to explain better.

The lab I work for wishes to optimize various machines, but they specifically wish to these optimization using solely analytical calculations for the sake of computational speed and intuition. We are curious as to whether Pyleecan is capable of this, so that we could avoid coding these analytical models on our own.

**Reply by gforti84** [reply]
If a may, you explanation is still not clear. There are several methods which may be called "analytical". There are some empirical formulations, Maxwell-Fourier method, magnetic equivalent circuit with reluctance networks, and so on.

**Reply by nkbw33** [reply]
We are currently using magnetic equivalent circuits with reluctance networks, but are open to alternative analytical methods as well.

**Reply by BonneelP** [reply]
For now pyleecan magnetic models are FEA only and pyleecan already includes optimization tools. Pyleecan can provides you the machine parametrization and some main methods/functions that you could reuse (compute resistances, compute slot surfaces, compute mmf...). In general all the geometrical aspect are handled in pyleecan and you can convert easily your machine to a polar equivalent. But I think you will have to implement yourself the overall method. Other advantage of doing that within pyleecan is that you will be able to discuss with the community about your implementation and you will be able to easily compare your results with FEMM to know if your computations are correct.

**Reply by nkbw33** [reply]
Oh, alright. We had thought differently based on one of the papers on Pyleecan we had read. Do you know what this quote is referring to:

> In Pyleecan OOP is used to define a simulation by combining several objects together. Each object is dedicated to a particular computation with a particular model/method. For instance, there are three Magnetic model objects in Pyleecan that compute the airgap flux in three different ways (Analytical, subdomain, FEA). It enables the user to know exactly how every step of the simulation is computed and to switch easily from one method to another. 

Found in this [paper](https://www.researchgate.net/publication/328520511_Pyleecan_An_Open-Source_Python_Object-Oriented_Software_for_the_Multiphysic_Design_Optimization_of_Electrical_Machines).

**Reply by BonneelP** [reply]
Yes we have chosen to focus on FEMM and drop the other models as FEMM is the most flexible one to handle all the geometrical complicity that pyleecan enables. A PR is open on the sub-model if you are interested: https://github.com/Eomys/pyleecan/pull/677

**Reply by nkbw33** [reply]
Alright, that makes sense. Is there anything left from the initial analytical model that you decided to drop that is accesible? 

Thank you for all your help.

**Reply by BonneelP** [reply]
The model was based on permeance/mmf, I think that few elementary methods to compute mmf are still there for instance.
